recipes/imgui/all/conanfile.py

```diff
@@ -42,6 +42,12 @@ def configure(self):
     def source(self):
         tools.get(**self.conan_data["sources"][self.version],
                   destination=self._source_subfolder, strip_root=True)
+        # Ensure we take into account export_headers
+        tools.replace_in_file(
```

Comment:
this sounds a bit intrusive. also it viiolates immutability of sources.

Reply:
This recipe does not do `no_copy_source`, so can't we move the patch step to a `_patch_sources` method, called at the start of `build`?

Reply:
This replacement is valid for all configurations, so all of them can use the same (modified) sources. Anyway, I like the underlying idea that `source()` retrieves just the sources to the cache, and then any modification is applied in the build directory.